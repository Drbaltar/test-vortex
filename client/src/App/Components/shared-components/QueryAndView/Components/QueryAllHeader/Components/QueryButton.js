import React from 'react';

const QueryButton = (props) => {
    const { hasQueryRan, query } = props;

    return (
        <div 
            className="btn btn-primary"
            style={{float: 'right'}}
            id="queryButton"
            onClick={() => query()}>
            {getButtonText(hasQueryRan)}
        </div>
    );
};

const getButtonText = (hasQueryRan) => {
    return hasQueryRan ? '&#8635; Refresh List' : 'Search';
};

export default QueryButton;