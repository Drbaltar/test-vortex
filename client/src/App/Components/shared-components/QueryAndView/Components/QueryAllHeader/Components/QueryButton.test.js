import React from 'react';
import { shallow } from 'enzyme';
import QueryButton from './QueryButton';

const mockQueryFunction = jest.fn();

describe('QueryButton', () => {
    const wrapper = shallow(<QueryButton hasQueryRan={false} query={mockQueryFunction}/>);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('sets the text of the button to `Search` if hasQueryRan prop is false', () => {
        wrapper.setProps({ hasQueryRan: false });
        expect(wrapper.find('.btn').text()).toBe('Search');
    });

    it('sets the text of the button to `Refresh List` if hasQueryRan prop is true', () => {
        wrapper.setProps({ hasQueryRan: true });
        expect(wrapper.find('.btn').text()).toBe('\u21BB Refresh List');
    });

    describe('when the user clicks on the button', () => {
        beforeAll(() => {
            wrapper.find('.btn').simulate('click');
        });

        it('calls the query function', () => {
            expect(mockQueryFunction).toHaveBeenCalled();
        });
    });
});