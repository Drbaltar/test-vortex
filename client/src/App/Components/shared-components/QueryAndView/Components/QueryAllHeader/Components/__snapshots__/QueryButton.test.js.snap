// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryButton renders correctly 1`] = `
<div
  className="btn btn-primary"
  id="queryButton"
  onClick={[Function]}
  style={
    Object {
      "float": "right",
    }
  }
>
  Search
</div>
`;
